PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = "mail.thread"
-    _inherit = "hr.employee"
-    _inherit = 'hr.employee.public'
-    _inherit = 'res.company'
-    _inherit = 'res.config.settings'

Total Changes: 16

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_attendance/models/hr_attendance.py
+++ b/addons/hr_attendance/models/hr_attendance.py
@@ -23,10 +23,9 @@ def get_google_maps_url(latitude, longitude):
 
 
 class HrAttendance(models.Model):
-    _name = "hr.attendance"
     _description = "Attendance"
     _order = "check_in desc"
-    _inherit = "mail.thread"
+    _inherit = ["mail.thread"]
 
     def _default_employee(self):
         return self.env.user.employee_id

--- a/addons/hr_attendance/models/hr_attendance_overtime.py
+++ b/addons/hr_attendance/models/hr_attendance_overtime.py
@@ -5,7 +5,6 @@ from odoo import models, fields
 
 
 class HrAttendanceOvertime(models.Model):
-    _name = "hr.attendance.overtime"
     _description = "Attendance Overtime"
     _rec_name = 'employee_id'
     _order = 'date desc'

--- a/addons/hr_attendance/models/hr_employee.py
+++ b/addons/hr_attendance/models/hr_employee.py
@@ -9,7 +9,7 @@ from odoo.tools import float_round
 
 
 class HrEmployee(models.Model):
-    _inherit = "hr.employee"
+    _inherit = ["hr.employee"]
 
     attendance_manager_id = fields.Many2one(
         'res.users', store=True, readonly=False,

--- a/addons/hr_attendance/models/hr_employee_base.py
+++ b/addons/hr_attendance/models/hr_employee_base.py
@@ -4,7 +4,7 @@ from odoo import api, models
 
 
 class HrEmployeeBase(models.AbstractModel):
-    _inherit = "hr.employee.base"
+    _inherit = ["hr.employee.base"]
 
     @api.depends("user_id.im_status", "attendance_state")
     def _compute_presence_state(self):

--- a/addons/hr_attendance/models/hr_employee_public.py
+++ b/addons/hr_attendance/models/hr_employee_public.py
@@ -3,8 +3,9 @@
 
 from odoo import fields, models
 
+
 class HrEmployeePublic(models.Model):
-    _inherit = 'hr.employee.public'
+    _inherit = ['hr.employee.public']
 
     # These are required for manual attendance
     attendance_state = fields.Selection(related='employee_id.attendance_state', readonly=True,

--- a/addons/hr_attendance/models/res_company.py
+++ b/addons/hr_attendance/models/res_company.py
@@ -8,7 +8,7 @@ from werkzeug.urls import url_join
 
 
 class ResCompany(models.Model):
-    _inherit = 'res.company'
+    _inherit = ['res.company']
 
     def _default_company_token(self):
         return str(uuid.uuid4())

--- a/addons/hr_attendance/models/res_config_settings.py
+++ b/addons/hr_attendance/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     overtime_company_threshold = fields.Integer(
         string="Tolerance Time In Favor Of Company", readonly=False)

--- a/addons/hr_attendance/models/res_users.py
+++ b/addons/hr_attendance/models/res_users.py
@@ -3,6 +3,7 @@
 
 from odoo import models, fields, _
 
+
 class User(models.Model):
     _inherit = ['res.users']
 
